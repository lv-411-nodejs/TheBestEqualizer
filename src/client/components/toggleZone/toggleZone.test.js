import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import ToggleZone from './toggleZone';

Enzyme.configure({ adapter: new Adapter() });

describe('Toggle Zone Component', () => {

    let wrapper;
    const setState = jest.fn();
    const useStateSpy = jest.spyOn(React, 'useState')
    useStateSpy.mockImplementation((init) => [init, setState]);

    const fakeChilds = {
        children: React.createElement('div')
    };

    beforeEach(() => {
        wrapper = shallow(<ToggleZone {...fakeChilds} />);
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should render ToggleZone component properly', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should handle onClick action on ToggleZone button', () => {
        wrapper.find('button').simulate('click');
        expect(wrapper.find('button').text()).toEqual('Hide Effects');
    });
});