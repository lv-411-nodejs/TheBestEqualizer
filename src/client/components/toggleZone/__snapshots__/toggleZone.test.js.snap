// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle Zone Component should render ToggleZone component properly 1`] = `
<Fragment>
  <button
    className="toggle__button"
    onClick={[Function]}
  >
    Show Effects
  </button>
  <div
    className="toggle "
  >
    <div />
  </div>
</Fragment>
`;
